external/corefx-bugfix/src/System.IO.FileSystem/src/System/IO/Enumeration/FileSystemEnumerator.Windows.cs

```diff
@@ -92,6 +92,23 @@ private void CloseDirectoryHandle()
         /// </summary>
         private IntPtr CreateDirectoryHandle(string path, bool ignoreNotFound = false)
         {
+            // GetFileAttributesEx sometimes does not understand long form UNC paths
+            // without adding a trailing backslash. So we check explicitly for such a
+            // path, and add the required trail.
+            if ((path.StartsWith(@"\?\") || path.StartsWith(@"\\?\"))
```

Comment:
Is there somewhere we can centralize this code so it only exists in one place? Right now there's two copies.

Reply:
@joshpeterson what do you think about this?

Reply:
I'm fine with this as-is, since this is a back port if the existing change on `unity-main`. It is not ideal, but I don't see a better solution.